plutus-core/plutus-ir/src/PlutusIR/Compiler/Definitions.hs

```diff
@@ -130,14 +135,32 @@ instance MonadDefs key uni fun ann m => MonadDefs key uni fun ann (ReaderT r m)
 defineTerm :: MonadDefs key uni fun ann m => key -> TermDefWithStrictness uni fun ann -> Set.Set key -> m ()
 defineTerm name def deps = liftDef $ DefT $ modify $ over termDefs $ Map.insert name (def, deps)
 
+modifyTermDef :: MonadDefs key uni fun ann m => key -> (TermDefWithStrictness uni fun ann -> TermDefWithStrictness uni fun ann)-> m ()
+modifyTermDef name f = liftDef $ DefT $ modify $ over termDefs $ Map.adjust (first f) name
+
 defineType :: MonadDefs key uni fun ann m => key -> TypeDef TyName uni ann -> Set.Set key -> m ()
 defineType name def deps = liftDef $ DefT $ modify $ over typeDefs $ Map.insert name (def, deps)
 
+modifyTypeDef :: MonadDefs key uni fun ann m => key -> (TypeDef TyName uni ann -> TypeDef TyName uni ann)-> m ()
+modifyTypeDef name f = liftDef $ DefT $ modify $ over typeDefs $ Map.adjust (first f) name
+
 defineDatatype
     :: forall key uni fun ann m . MonadDefs key uni fun ann m
     => key -> DatatypeDef TyName Name uni fun ann -> Set.Set key -> m ()
 defineDatatype name def deps = liftDef $ DefT $ modify $ over datatypeDefs $ Map.insert name (def, deps)
 
+modifyDatatypeDef :: MonadDefs key uni fun ann m => key -> (DatatypeDef TyName Name uni fun ann -> DatatypeDef TyName Name uni fun ann)-> m ()
+modifyDatatypeDef name f = liftDef $ DefT $ modify $ over datatypeDefs $ Map.adjust (first f) name
+
```

Comment:
Not critical but these have some repetition. I see that HOF is not possible here because `termDefs` and `typeDefs` etc have different types. But e.g., would it make sense to have a function (I didn't give it a good name):

```haskell
modifyF :: MonadDefs key uni fun ann m =>
    (DefState key uni fun ann -> DefState key uni fun ann) -> m ()
modifyF f = liftDef $ DefT $ modify f
```
and instead of the repeated `liftDef $ DefT $ modify` in all these functions we have `modifyF` in the beginning instead.

Reply:
Yeah, it's dumb, but I didn't think it was worth factoring it out here, it's mechanical enough :shrug:

Reply:
Fair enough.